### PR TITLE
DocFX documentation

### PR DESCRIPTION
**Pull request by dealloc**
this PR adds [DocFX](https://dotnet.github.io/docfx) documentation capabilities.

notable features in this PR:
- removal of `UseStaticFiles` from API
  - This removes a middleware step, thus reducing the amount of operations for *each* request
  - OpenAPI spec no longer drains bandwidth from API as it's served from GH pages
- adds `docs` folder under which extensive documentation can be added (for example the OpenAPI specs)
- automatically generated API docs (API as in C# code) to help contributors get a grasp on the codebase
- support for markdown, diagrams, videos, images, ... in documentation

One current issue (#46) essentially needs this setup so we have a place to put Docker documentation etc.

While the documentation in it's current form is bare bones (I'm open to not merge this PR until it's fleshed out further, by me or other maintainers) it provides a framework for taking the documentation to the next level.

PS documentation also provides dark mode out of the box <3

### CODE REVIEW

File: index.md
Comment (chatterchats) [inline index.md:1]: I think for initial implementation should have something other than boilerplate

**Review by chatterchats — COMMENTED** [review]
Looks pretty good overall, I think some of the files could be fleshed out a little bit more for the initial implementation of DocFX, however I don't think it's a reason to hold up deployment so that others may work on it if they wish

File: docs/openapi/swagger-ui.html
Comment (lambstream) [inline docs/openapi/swagger-ui.html:1]: Do we still need the swagger ui?

Comment (lambstream) [reply docs/openapi/swagger-ui.html:1]: As far as i can see it's no longer being hosted anywhere? Or is it being used internally by DocFX?

Comment (lambstream) [reply docs/openapi/swagger-ui.html:1]: Oh I see, alright then!

**Review by lambstream — APPROVED** [review]
Looks good to me

**Pull request merged**